webrtc/README.md

```diff
@@ -161,6 +165,51 @@ MUST pass an empty string. When receiving an `RTCDataChannel` via
 an empty string. This allows future versions of this specification to make use
 of the `RTCDataChannel` `label` property.
 
+## Closing an `RTCDataChannel`
+
+Some WebRTC implementations do not guarantee that any queued messages will be
+sent after a datachannel is closed.  Other implementations maintain separate
+outgoing message and transport queues, the status of which may not be visible
+to the user. Consequently we must add an additional layer of signaling to ensure
+reliable data delivery.
+
+When a node wishes to close a stream for writing, it SHOULD send a message with
+the `FIN` flag set, then proceed to wait for a `FIN_ACK` message from the remote
+node.
+
+If a `FIN` flag is received the node SHOULD respond with a `FIN_ACK`.
+
+When a `FIN_ACK` and a `FIN` have been received, a node may close the datachannel.
+
+The node MAY close the datachannel without receiving a `FIN_ACK`, for example in
+the case of a timeout, but there will be no guarantee that all previously sent
+messages have been received.
+
+If a node has previously sent a `STOP_SENDING` flag to the remote node, it MUST
+continue to act on any flags present in received messages in order to
+successfully process an incoming `FIN_ACK`.
+
+### Example of closing an `RTCDataChannel`
+
+NodeA closes for writing, NodeB delays allowing the channel to close until it also
+finishes writing.
+
+```mermaid
+sequenceDiagram
+    NodeA->>NodeB: DATA
+    NodeA->>NodeB: FIN
+    NodeB->>NodeA: FIN_ACK
+    NodeB->>NodeA: DATA
+    NodeB->>NodeA: FIN
+    NodeA->>NodeB: FIN_ACK
+```
+
+After NodeA has received the `FIN` it is free to close the datachannel since it
+has previously received a `FIN_ACK`. If NodeB receives the `FIN_ACK` before this
+it may close the channel since it previously received a `FIN`.
+
+This way the channel can be closed from either end without data loss.
```

Comment:
```suggestion
After _A_ has received the `FIN` it is free to close the datachannel since it
has previously received a `FIN_ACK`. If _B_ receives the `FIN_ACK` before this
it may close the channel since it previously received a `FIN`.

This way the channel can be closed from either end without data loss.
```

I think other parts in this document just use italic for the identifiers.